Source/DafnyLanguageServer/Caching/ByteArrayEquality.cs

```diff
@@ -0,0 +1,34 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Policy;
+
+namespace Microsoft.Dafny.LanguageServer.Language;
+
+class ByteArrayEquality : IEqualityComparer<byte[]> {
+  public int Compare(byte[]? x, byte[]? y) {
+    return (x, y) switch {
+      (null, null) => 0,
+      (null, _) => 1,
+      (_, null) => -1,
+      _ => x.SequenceEqual(y) ? 0 : 1
+    };
```

Comment:
1. It's a bit strange that if both `x` and `y` are both non-null and unequal, then both `Compare(x, y)` and `Compare(y, x)` return 1.
2. `Compare` is defined not in `IEqualityComparer<T>` but instead in `IComparer<T>`, did you mean to also add the latter to the list of derived interfaces?

Reply:
Removed the `Compare` method, it wasn't used.